src/node/BrowserRunner.ts

```diff
@@ -199,6 +192,14 @@ export class BrowserRunner {
         );
       }
     }
+
+    // Attempt to remove temporary profile directory to avoid littering.
```

Comment:
What if this.proc.kill('SIGKILL') throws but the process is still killed? Can that happen in practice? That might be the reason for removing the temp dir upfront?

Reply:
Good question. So I've taken a look at the [subprocess.kill()](https://nodejs.org/api/child_process.html#subprocesskillsignal) documentation, and I cannot see that an error gets thrown at all. So my question would be do we really need the try/catch and will we ever re-throw the error (that is the part that I totally issed - thanks for noting!).

Shouldn't a check for the return value be performed instead? We could then throw an error at the end of the method, and could still have the user data directory deletion before. If it is `false` we could still try to remove the user data directory even if not fully successful.

Maybe the [`error`](https://nodejs.org/api/child_process.html#event-error) and [`close`](https://nodejs.org/api/child_process.html#event-close) events could be used here to better handle the kill status by making the `BrowserRunner.kill()` method async.

Reply:
Yeah, that's strange. I could not find why it was like this in the first place.  It does not look like it can throw https://github.com/nodejs/node/blob/master/lib/child_process.js#L419